lib/active_merchant/billing/gateways/payu_latam.rb

```diff
@@ -0,0 +1,484 @@
+          shipping_address[:street1] = address[:address1]
+          shipping_address[:street2] = address[:address2]
+          shipping_address[:city] = address[:city]
+          shipping_address[:state] = address[:state]
+          shipping_address[:country] = address[:country]
+          shipping_address[:postalCode] = address[:zip]
+          shipping_address[:phone] = address[:phone]
+          post[:transaction][:order][:shippingAddress] = shipping_address
+        end
+      end
+
+      def add_invoice(post, money, options)
+        tx_value = {}
+        tx_value[:value] = amount(money)
+        tx_value[:currency] = options[:currency] || currency(money)
+
+        tx_tax = {}
+        tx_tax[:value] = '0'
+        tx_tax[:currency] = options[:currency] || currency(money)
+
+        tx_tax_return_base = {}
+        tx_tax_return_base[:value] = '0'
+        tx_tax_return_base[:currency] = options[:currency] || currency(money)
+
+        additional_values = {}
+        additional_values[:TX_VALUE] = tx_value
+        additional_values[:TX_TAX] = tx_tax
+        additional_values[:TX_TAX_RETURN_BASE] = tx_tax_return_base
+
+        post[:transaction][:order][:additionalValues] = additional_values
+      end
+
+      def add_signature(post)
+        post[:transaction][:order][:signature] = signature_from(post)
+      end
+
+      def signature_from(post)
+        signature_string = [
+          @options[:api_key],
+          @options[:merchant_id],
+          post[:transaction][:order][:referenceCode],
+          post[:transaction][:order][:additionalValues][:TX_VALUE][:value],
+          post[:transaction][:order][:additionalValues][:TX_VALUE][:currency]
+        ].compact.join("~")
+
+        Digest::MD5.hexdigest(signature_string)
+      end
+
+      def add_payment_method(post, payment_method, options)
+        if payment_method.is_a?(String)
+          _, brand, token = split_authorization(payment_method)
+          credit_card = {}
+          credit_card[:securityCode] = options[:cvv] if options[:cvv]
+          post[:transaction][:creditCard] = credit_card
+          post[:transaction][:creditCardTokenId] = token
+          post[:transaction][:paymentMethod] = brand.upcase
+        else
+          credit_card = {}
+          credit_card[:number] = payment_method.number
+          credit_card[:securityCode] = payment_method.verification_value
+          credit_card[:expirationDate] = format(payment_method.year, :four_digits).to_s + '/' + format(payment_method.month, :two_digits).to_s
+          credit_card[:name] = payment_method.name.strip
+          post[:transaction][:creditCard] = credit_card
+          post[:transaction][:paymentMethod] = BRAND_MAP[payment_method.brand.to_s]
+        end
+      end
+
+      def add_payer(post, options)
+        payer = {}
+        payer[:dniNumber] = options[:payer_dni_number] if options[:payer_dni_number]
+        payer[:cnpj] = options[:payer_cnpj] if options[:payer_cnpj]
```

Comment:
I'm not sure what `dniNumber` and `payer_cnpj` represent or how likely customers are to use them.  Do you think it's likely someone might want them?